src/util-logopenfile.c

```diff
@@ -516,6 +870,12 @@ int LogFileFreeCtx(LogFileCtx *lf_ctx)
             SCFree(lf_ctx->redis_setup.key);
     }
 #endif
+#ifdef HAVE_LIBRDKAFKA
+
+    if (lf_ctx->type == LOGFILE_TYPE_KAFKA) {
+        SCLogFileCloseKafka(lf_ctx);
```

Comment:
Should this be done under lock? I see redis doesn't do it, but that may be a mistake.